Client/Frontend/Passcode/PasscodeMigrationView.swift

```diff
@@ -0,0 +1,129 @@
+// Copyright 2021 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import SwiftUI
+import BraveUI
+import LocalAuthentication
+import Shared
+
+struct PasscodeMigrationContainerView: View {
+    var dismiss: ((_ enableBrowserLock: Bool) -> Void)?
+    private let isUserAuthenticationAvailable: Bool
+    private let biometryType: LABiometryType
+    private let context: LAContext
+    
+    init() {
+        context = LAContext()
+        isUserAuthenticationAvailable = context
+            .canEvaluatePolicy(.deviceOwnerAuthentication, error: nil)
+        biometryType = context.biometryType
+    }
+    
+    var body: some View {
+        PasscodeMigrationView(
+            isUserAuthenticationAvailable: isUserAuthenticationAvailable,
+            availableBiometryMethod: biometryType
+        ) {
+            if isUserAuthenticationAvailable {
```

Comment:
We should do the evaluation here as well because `canEvaluatePolicy` states not to store the value for any duration of time because the policy can change. For example, the user might remove their fingerprints or passcode while this screen is active or the info might be invalidated by the system at any moment.

https://developer.apple.com/documentation/localauthentication/lacontext/1514149-canevaluatepolicy

`Don’t store the return value from this method because it might change as a result of changes in the system.`

Reply:
Fixed